pkg/kvstore/lock.go

```diff
@@ -58,9 +65,9 @@ func (pl *pathLocks) lock(path string) {
 			return
 		}
 
-		if time.Since(started) > lockTimeout {
+		if time.Since(started) > staleLockTimeout {
 			log.WithField("path", path).Warning("WARNING: Timeout while waiting for lock, ignoring lock")
-			pl.lockPaths[path] = 1
+			pl.lockPaths[path] = 0
 			pl.mutex.Unlock()
```

Comment:
@tgraf how will the caller distinguish that the lock was really gotten or not?

Reply:
I've changed it to restart the acquisition cycle. The downside of this is a theoretical chance of permanent starvation but handles the case when multiple local consumers are waiting on the same key.